XTagInfoProducer/plugins/XTagInfoProducer.cc

```diff
@@ -472,24 +471,31 @@ XTagInfoProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
             {
                 const pat::Electron & electron = *findElectron->second;
                 cpf_features.cpf_matchedElectron = 1;
+		if(reco::deltaR(electron , jet) > 0.4 ) continue ; 
 
                 elec_features.elec_ptrel = electron.pt()/uncorrectedPt;
                 elec_features.elec_deta = std::fabs(electron.eta()-jet.eta());
                 elec_features.elec_dphi = std::fabs(reco::deltaPhi(electron.phi(),jet.phi())); 
                 elec_features.elec_charge = electron.charge(); 
                 elec_features.elec_energy = electron.energy()/electron.pt(); 
-                elec_features.elec_jetDeltaR = reco::deltaR(electron , jet); 
+                elec_features.elec_jetDeltaR = reco::deltaR(electron,jet); 
                 elec_features.elec_EtFromCaloEn = electron.caloEnergy() * sin(electron.p4().theta());
                 elec_features.elec_ecalDrivenSeed = electron.ecalDrivenSeed();
 
                 elec_features.elec_isEB = electron.isEB();
                 elec_features.elec_isEE  = electron.isEE();
                 elec_features.elec_ecalEnergy  = electron.ecalEnergy()/electron.pt();
                 elec_features.elec_isPassConversionVeto = electron.passConversionVeto();
-
+		if(electron.convDist() != -9999){
```

Comment:
note that convDist() returns float so the comparison to int here is dangerous; better 
`if (std::fabs(electron.convDist()+9999)<1e-4)) `
if you want to test for -9999 only; if any negative value make no sense just use
`if (electron.convDist()<0) `
instead